pkg/services/slack.go

```diff
@@ -96,14 +97,18 @@ func (n *SlackNotification) GetTemplater(name string, f texttemplate.FuncMap) (T
 }
 
 type SlackOptions struct {
-	Username           string   `json:"username"`
-	Icon               string   `json:"icon"`
-	Token              string   `json:"token"`
-	SigningSecret      string   `json:"signingSecret"`
-	Channels           []string `json:"channels"`
-	InsecureSkipVerify bool     `json:"insecureSkipVerify"`
-	ApiURL             string   `json:"apiURL"`
-	DisableUnfurl      bool     `json:"disableUnfurl"`
+	Username            string        `json:"username"`
+	Icon                string        `json:"icon"`
+	Token               string        `json:"token"`
+	SigningSecret       string        `json:"signingSecret"`
+	Channels            []string      `json:"channels"`
+	ApiURL              string        `json:"apiURL"`
+	DisableUnfurl       bool          `json:"disableUnfurl"`
+	InsecureSkipVerify  bool          `json:"insecureSkipVerify"`
+	MaxIdleConns        int           `json:"maxIdleConns"`
+	MaxIdleConnsPerHost int           `json:"maxIdleConnsPerHost"`
+	MaxConnsPerHost     int           `json:"maxConnsPerHost"`
+	IdleConnTimeout     time.Duration `json:"idleConnTimeout"`
```

Comment:
We can't use time.Duration for parsing time values - we will need to use a string and then parse the actual `time.Duration` by doing something like this 

```golang
	idleConnTimeout, err := time.ParseDuration(s.opts.IdleConnTimeout)
	if err != nil {
		return errors.Wrap(err, "failed to parse idle connection timeout")
	}
```

and then passing in `idleConnTimeout` to the `NewTransport` function.